lab4/provided/processor.hh

```diff
@@ -6,28 +6,23 @@
 #include <map>
 #include <queue>
 #include <utility>
+#include <string>
 
 #include "cache.hh"
 #include "ticked_object.hh"
+#include "record_store.hh"
 
 class Processor: public TickedObject
 {
-  private:
+  protected:
     Cache *cache;
     Memory *memory;
 
-    struct Record {
-        int64_t ticksFromNow;
-        bool write;
-        uint64_t address;
-        int requestId;
-        int size;
-        uint8_t* data;
-    };
+    RecordStore *records;
 
-    std::queue<Record> trace;
+    std::queue<Record*> trace;
```

Comment:
I would rather see references, but you don't have to go back and change everything. This is OK as it is.